### PR TITLE
Upgrade capi dependencies

### PR DESCRIPTION
**Pull request by njuettner**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by whites11 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Upgrade CAPi dependencies.
```
[the remark this answers is not in the document]
Comment (njuettner) [reply CHANGELOG.md]: Thanks buddy ❤️

File: service/service.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func New(config Config) (*Service, error) {
 	{
 		c := k8sclient.ClientsConfig{
 			SchemeBuilder: k8sclient.SchemeBuilder{
-				apiv1alpha3.AddToScheme,
+				capi.AddToScheme,
```
Comment (fiunchinho) [inline service/service.go:83]: Could we even remove the `cluster-api``dependency? I don't think it's used?

Comment (njuettner) [reply service/service.go:83]: This makes sense, I didn't check where it's been used. You're right it's not even used 🙈. I dropped it.

File: go.mod [hunk 3]
```diff
@@ -1,34 +1,98 @@
...
+	sigs.k8s.io/structured-merge-diff/v4 v4.1.2 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
 replace (
 	github.com/coreos/etcd v3.3.10+incompatible => github.com/coreos/etcd v3.3.25+incompatible
 	github.com/coreos/etcd v3.3.13+incompatible => github.com/coreos/etcd v3.3.25+incompatible
 	github.com/dgrijalva/jwt-go => github.com/dgrijalva/jwt-go/v4 v4.0.0-preview1
 	github.com/gogo/protobuf v1.3.1 => github.com/gogo/protobuf v1.3.2
-	sigs.k8s.io/cluster-api => github.com/giantswarm/cluster-api v0.3.13-gs
```
Comment (fiunchinho) [inline go.mod:33]: every time I removed this from other repos, the build failed because of capz dependency on `apiextensions`

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Upgrade CAPI dependencies.
```
Comment (fiunchinho) [inline CHANGELOG.md:12]:
```suggestion
- Upgrade apiextensions to `v6.0.0`.
- Remove `cluster-api` dependency.
```

What do you think?

File: go.mod (hunk 3 above)
Comment (njuettner) [reply go.mod:33]: Ok will add it again 🚀

File: CHANGELOG.md (hunk 4 above)
Comment (njuettner) [reply CHANGELOG.md:12]: That sounds good 👍🏻

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**